electrode_models/conversion_electrode.py

```diff
@@ -1,114 +1,188 @@
-
-    Three phase boundaries are not currently implemented, but certainly could be, at some later date.
+    Class file for air electrode methods
 """
 
 import cantera as ct
-from math import tanh
+from math import tanh, pi, isnan, exp
 import numpy as np
+import matplotlib
 
-class electrode(): 
+class electrode():
     """
-    Create an electrode object representing the single particle electrode.
+    Create an electrode object representing the air electrode.
     """
 
-    def __init__(self, input_file, inputs, sep_inputs, counter_inputs,    
+    def __init__(self, input_file, inputs, sep_inputs, an_inputs,
         electrode_name, params, offset):
         """
         Initialize the model.
-        """ 
-   
+        """
+
         # Import relevant Cantera objects.
         self.elyte_obj = ct.Solution(input_file, inputs['electrolyte-phase'])
         self.host_obj = ct.Solution(input_file, inputs['host-phase'])
-        self.host_surf_obj = ct.Interface(input_file, inputs['surf-phase'], 
+        self.host_surf_obj = ct.Interface(input_file, inputs['surf-phase'],
             [self.elyte_obj, self.host_obj])
-       
+
+        self.C_k_0 = [species['C_k'] for species in sep_inputs['transport']['diffusion-coefficients']]
+
         # Create conversion phases and associated electrolyte interfaces:
         self.conversion_phases = []
         self.conversion_obj = []
         self.conversion_surf_obj = []
+        self.conversion_tpb_obj = []
         self.n_conversion_phases = 0
+        self.conv_ph_min = []
         for ph in inputs["conversion-phases"]:
             self.conversion_phases.append(ph["bulk-name"])
             self.conversion_obj.append(ct.Solution(input_file, ph["bulk-name"]))
-            self.conversion_surf_obj.append(ct.Interface(input_file, 
-                ph["surf-name"],[self.elyte_obj, self.host_obj, 
+            self.conversion_surf_obj.append(ct.Interface(input_file,
+                ph["surf-name"],[self.elyte_obj, self.host_obj,
                 self.conversion_obj[self.n_conversion_phases]]))
+            if ph["tpb-name"]:
+                self.conversion_tpb_obj.append(ct.Interface(input_file,
+                    ph["tpb-name"], [self.elyte_obj, self.host_obj,
+                    self.conversion_obj[self.n_conversion_phases]]))
+            self.conv_ph_min.append(ph["min-vol-frac"])
             self.n_conversion_phases += 1
 
-        # Anode or cathode? Positive external current delivers positive charge 
-        # to the anode, and removes positive charge from the cathode.
+        E0_ca = self.host_surf_obj.delta_standard_gibbs/ct.faraday
+        print('E_eq_ca =', -E0_ca)
+
+        self.sw_conv = np.ones((self.n_conversion_phases))
+
+        # Electrode thickness and inverse thickness:
+        self.n_points = inputs['n-points']
+        print('Cathode points =', self.n_points)
```

Comment:
Here too :)

Actually, probably better to just remove the print, while we're at it.